Add docs for initial scale (#2720)
knative/docs: docs/serving/autoscaling/scale-bounds.md

To apply upper and lower bounds to autoscaling behavior, you can specify scale bounds in both directions.

You can also specify the initial target scale of a revision after creation, with either its cluster-wide flag or annotation.

> If both `minScale` annotation and initial scale (global or per-revision) are specified, and `minScale` is greater than the calculated initial scale, the initial scale will be ignored.

Reviewer: ```suggestion
If both `minScale` and initial scale annotations (global or per-revision) are specified, and `minScale` is greater than the calculated initial scale, the initial scale will be ignored.
```

Author: I just updated this to make it more explicit (instead of saying "global or per-revision" here)

> To apply upper and lower bounds to autoscaling behavior, you can specify scale bounds in both directions.

Reviewer 2: ```suggestion
You can configure upper and lower bounds to control autoscaling behavior.
```

> You can also specify the initial target scale of a revision after creation, with either its cluster-wide flag or annotation.

Reviewer 3: ```suggestion
You can also specify the initial scale a revision is scaled to immediately after creation, either as a default for all revisions, or for a specific revision using an annotation.
```

> You can also specify the initial scale a revision is scaled to immediately after creation, either as a default for all revisions, or for a specific revision using an annotation.

Reviewer 2: ```suggestion
You can also specify the initial scale that a Revision is scaled to immediately after creation.
This can be a default configuration for all Revisions, or for a specific Revision using an annotation.
```

## Lower bound

> This value controls the minimum number of replicas that each revision should have.

Reviewer 2: ```suggestion
This value controls the minimum number of replicas that each Revision should have.
```

You can also specify the initial target scale of a revision after creation, with either its cluster-wide flag or annotation.

> If both `minScale` annotation and `initial-scale` ConfigMap value or `autoscaling.knative.dev/initialScale` annotation are specified, and `minScale` is greater than the calculated initial scale, the initial scale will be ignored.

Reviewer: ```suggestion
If the `minScale` annotation and the `autoscaling.knative.dev/initialScale` annotation are both specified, and `minScale` is greater than the calculated initial scale, the initial scale will be ignored.
```

Author: I just moved this sentence down because it makes more sense there. I also decided not to mention annotation, because it's not entirely accurate, since it's the combination of annotation value and cluster-wide value that determines the calculated initial scale.

{{< /tab >}}
{{< /tabs >}}

## Initial scale

> This value controls the initial target scale of a revision after creation. If lower bound is set and is greater than the initial scale, the initial scale will be ignored.

Reviewer 3: I think this could use an extra sentence to describe what initial scale actually does. Maybe something like:

```suggestion
This value controls the initial target scale of a revision after creation. This is the target scale a revision must reach, immediately after first creation, before being marked Ready. Once the revision has reached this scale once, this value is ignored.

If a lower bound is set, the revision must achieve this scale before being marked Ready, so if it is greater than the initial scale, the initial scale will effectively be ignored.
```

Author: Yes this would be very helpful. Should we also mention the revision will be scale down immediately after the initial target scale is reached if the received traffic only needs a smaller scale?

Reviewer 3: Depends if we think "Once the revision has reached this scale once, this value is ignored" is enough for readers to guess this.. I think it might be, but also I could see any extra sentence being helpful to spell it out.

{{< /tab >}}
{{< /tabs >}}

## Initial scale

> This value controls the initial target scale of a revision after creation. This is the target scale a revision must reach, immediately after creation, before being marked Ready. Once the revision has reached this scale once, this value is ignored (e.g. the revision will scale down after the initial target scale is reached if the traffic received only needs a smaller scale).

Reviewer 2: ```suggestion
This value controls the initial target scale a Revision must reach immediately after it is created before it is marked as `Ready`.
After the Revision has reached this scale one time, this value is ignored. This means that the Revision will scale down after the initial target scale is reached if the actual traffic received only needs a smaller scale.
```

> This value controls the initial target scale of a revision after creation. This is the target scale a revision must reach, immediately after first creation, before being marked Ready. Once the revision has reached this scale once, this value is ignored (e.g. the revision will scale down after the initial target scale is reached if the traffic received only needs a smaller scale).

Reviewer 4: ```suggestion
This value controls the initial target scale of a revision after creation. This is the target scale a revision must reach, immediately after creation, before being marked Ready. Once the revision has reached this scale once, this value is ignored (e.g. the revision will scale down after the initial target scale is reached if the traffic received only needs a smaller scale).
```
I think creation is _first_ by definition :)

> If lower bound is set and is greater than the initial scale, the initial scale will be ignored.

Reviewer 4: Perhaps we can rephrase like:
```
During creation the revision the larger of initial scale and lower bound is picked.
```

> During the creation of the revision, the larger of initial scale and lower bound is picked as the target scale.

Reviewer 2: ```suggestion
When the Revision is created, the larger of initial scale and lower bound is automatically chosen as the target scale.
```

## Initial scale

This value controls the initial target scale of a revision after creation. This is the target scale a revision must reach, immediately after creation, before being marked Ready. Once the revision has reached this scale once, this value is ignored (e.g. the revision will scale down after the initial target scale is reached if the traffic received only needs a smaller scale).

> During creation the revision, the larger of initial scale and lower bound is picked as the target scale.

Reviewer 4: ```suggestion
During the creation of the revision, the larger of initial scale and lower bound is picked as the target scale.
```
My text might have not been very precise :D
